### DISCUSSION TITLE
Keep enemy unit on map (don't despawn/virtualize unit)

### DISCUSSION THREAD

**Discussion by SgtKrunch** [Vdauphin/HeartsAndMinds#1613] [Question and answer] [answered]
I'm trying to keep a SAM system on the map, but part of it got initialized by H&M and will now only spawn back in if someone is within the city range. How do I keep specific units from being initialized into the system? (If possible)

**Comment by SgtKrunch** [comment]
What I've tried so far to no success is changing the side of the SAM/Radar to independent (current enemy faction is opfor), and assigning the following variables to them:

`this setVariable ["acex_headless_blacklist", true];
this setVariable ["no_cache", true];`

**Reply by Vdauphin** [reply]
Hello,

>  `this setVariable ["no_cache", true];`

I would expect this to work
Do you apply it to the group?

Here is the piece of code which delete group in range of city: https://github.com/Vdauphin/HeartsAndMinds/blob/c34896aa3889efb9b02dab882af8aa2b2c9547a1/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/city/de_activate.sqf#L46-L58

Cheers

**Reply by SgtKrunch** [reply]
> > `this setVariable ["no_cache", true];`
> 
> I would expect this to work Do you apply it to the group?
> 

Since it's technically a drone, i'm not sure how to apply it to it's group. I was hoping applying it to the object would work. i'll look into how to apply things to a drone's group

**Reply by Vdauphin** [reply]
`(group driver this) setVariable ["no_cache", true];` ?

**Reply by SgtKrunch** [reply]
Now I'm really confused. I've got it working on one map, but not another. Both maps/missions are pretty much identical except for the `param.hpp` file and the map itself. The one that is working for me currently is Lythium, the one that isn't working is Altis. I've made sure to test both maps on a dedicated server, and the result is the same. Altis, for whatever reason, does not want to work. This is the code that I've set in the init of the radar:
```
[radar_01] joinSilent (createGroup [independent, true]);

radar_01 setVariable ["no_cache", ["1", true], owner driver radar_01];
radar_01 setVariable ["acex_headless_blacklist", ["1", true], owner driver radar_01];
```
I've also set it to respawn, using the vehicle respawn module with a similar expression, just with `radar_01 = _this select 0;` at the start.

This works on Lythium, but not for Altis. Same exact setup. Only difference, other than location, is how many zones overlap the assets. On Lythium, only 2 zones overlap the radar. on Altis, it's more like 3 or 4 zones. Idk if the amount of zones overlapping has anything to do with it or not.

**Reply by Vdauphin** [reply, marked as answer]
Just use this in the init field:
```sqf
(group this) setVariable ["no_cache", true];
```

**Reply by Vdauphin** [reply]
**Please** don't use Lythium, it is a broken map, more than a half of city have spawn radius under 100 and even 15 meter spawn radius
https://github.com/Vdauphin/HeartsAndMinds/pull/753

**Comment by SgtKrunch** [comment]
> Just use this in the init field:
> 
> ```sqf
> (group this) setVariable ["no_cache", true];
> ```

It worked lol XD, I don't know why I couldn't find that solution sooner, i wasted so many hours of my life trying to fix that lol.


> **Please** don't use Lythium, it is a broken map, more than a half of city have spawn radius under 100 and even 15 meter spawn radius #753

We just finished a HAM campaign on it, didn't really give us any issues. Did crank the radius offset by like 800, but still worked pretty well.

**Reply by Vdauphin** [reply]
This only change how far player activate city not how H&M populate city which depend on the radius gave by the map author
Low spawn radius decrease a lot the number of enemies and civi because H&M don't find houses and city looks dead
